components/src/dynamo/common/utils/prometheus.py

```
@@ -120,6 +120,18 @@ def register_engine_metrics_callback(
             labels.ENDPOINT: endpoint_name_final,  # "dynamo_endpoint"
         }
 
+        # Add worker_id label from connection_id (discovery instance ID).
+        # This provides a stable per-worker identity label so metrics from different
+        # workers serving the same endpoint can be distinguished without relying on
+        # Kubernetes labels. Mirrors Rust auto-label injection in create_metric().
+        try:
+            conn_id = endpoint.connection_id()
+            auto_labels[labels.WORKER_ID] = format(conn_id, "x")
+        except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)